src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_1_TypicalWorkflow
+    {
+        [Workflow(runMethod: nameof(SayHelloAsync))]
+        public class SayHelloWorkflow
+        {
+            private const string AddresseeNameDefault = "Boss";
+
+            private string _addresseeName = null;
+
+            public async Task SayHelloAsync(WorkflowContext workflowCtx)
+            {
+                var greeting = new SpeechRequest($"Hello, {_addresseeName ?? AddresseeNameDefault}.");                
+                await workflowCtx.Orchestrator.Activities.ExecuteAsync("SpeakAGreeting", greeting);
+            }
+
+            [WorkflowSignalHandler]
+            public Task SetAddresseeAsync(SpeechRequest input, WorkflowContext _)
```

Comment:
Should this be marked async? Or if we're gonna allow them to have non-async, why even use `Task` here (we can wrap that internally when we call a non-async func)?

Reply:
async just tells the compiler to allow `await` in this method. Task is about saying that the method is asynchronous. There are many ways a method can be asynchronous without using the await keyword.